solutions/CF Save the Nature.mdx

```diff
@@ -0,0 +1,89 @@
+---
+id: cf-1223C
+source: CF
+title: Save the Nature
+author: Jesse Choe, Kevin Sheng
+---
+
+[Official Editorial](https://codeforces.com/blog/entry/66993)
+
+Binary search on the answer for the minimum number of tickets, which we call $t$, to sell to make the total ecological contribution to at least $k$. 
+
+To check if a certain number of tickets reaches a minimal contribution of $k$, we can greedily observe that in order to minimize the number of tickets sold, we should maximize the total revenue from selling these tickets. If, however, it is impossible to sell at least $k$ tickets after selling all tickets, then output $-1$.
+
+Additionally, the problem indicates that only $x\%$ of the $a$-th ticket and $y\%$ of the $b$-th ticket could contribute to the total ecological contribution. Thus, we can iterate over every $a$-th ticket and $b$-th ticket at most $t$ and add $x\%$ to the total percentages contributing to an ecological program for each $a$-th ticket and $y\%$ for each $b$-th ticket, which we can store in $pct$.
+
+It suffices to sort the tickets and percentages in decreasing order. Our total ecological contribution is $\sum\limits_{i=1}^{t} p_i \cdot pct_i$.
+
+All that remains is to check whether the total sum of ecological contribution is at least $k$, which can be done in constant time.
+
+# Solution 
+
+**Time Complexity:** $\mathcal{O}(N\log^2 N)$
```

Comment:
maybe add the amount of queries into the complexity? not sure if needed tho

Reply:
Not needed since it says that the total sum of $N$ doesn't exceed 2*10^5